src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -518,6 +529,23 @@ public void NormalizeGraphPosition()
             }
         }
 
+        /// <summary>
+        /// To shift the whole graph back to its original position.
+        /// </summary>
+        public void SetGraphPosition()
+        {
+            double moveX = CenterX - (Nodes.Max(n => n.X + n.Width) - Nodes.Min(n => n.X)) / 2;
```

Comment:
We could probably refactor these expressions: `(Nodes.Max(n => n.X + n.Width) - Nodes.Min(n => n.X)) / 2;` in a property or method and call that in the `SetGraphPosition` and `RecordInitialPosition` methods?